### PR TITLE
negar/knowledge_test

### PR DESCRIPTION
**Pull request by negarn**
refactor knowledge test

### CODE REVIEW

File: src/javascript/binary_japan/knowledge_test/knowledge_test.init.js [hunk 1]
```diff
@@ -1,186 +1,222 @@
...
         $('#knowledge-test-msg').text(localize(msg));
         $('#knowledge-test-instructions').addClass('invisible');
     };
 
-    const showDisallowedMsg = function(jpStatus) {
-        const msgTemplate =
-            '{JAPAN ONLY}Dear customer, you are not allowed to take knowledge test until [_1]. Last test taken at [_2].';
-
-        const msg = localize(msgTemplate, [
+    const showDisallowedMsg = jpStatus =>
```
Comment (kellybinary) [inline src/javascript/binary_japan/knowledge_test/knowledge_test.init.js]: missing underscore here :)

File: src/javascript/binary_japan/knowledge_test/knowledge_test.init.js [hunk 2]
```diff
@@ -1,186 +1,222 @@
...
+                category: question_info.category,
+                id      : question_info.id,
+                question: question_info.question,
+                answer  : question_info.answer ? 1 : 0,
                 pass    : score,
             });
         });
-        KnowledgeTestData.sendResult(questions, resultScore);
-        submitCompleted = true;
+        sendResult(questions);
```
Comment (kellybinary) [inline src/javascript/binary_japan/knowledge_test/knowledge_test.init.js]: why resultScore is not being pass here?

File: src/javascript/binary_japan/knowledge_test/knowledge_test.init.js (hunk 1 above)
Comment (negarn) [inline src/javascript/binary_japan/knowledge_test/knowledge_test.init.js]: thanks, I will change it

File: src/javascript/binary_japan/knowledge_test/knowledge_test.init.js (hunk 2 above)
Comment (negarn) [inline src/javascript/binary_japan/knowledge_test/knowledge_test.init.js]: it's global

Comment (kellybinary) [inline src/javascript/binary_japan/knowledge_test/knowledge_test.init.js]: nice!

**Pull request merged**